common/scala/src/main/scala/org/apache/openwhisk/utils/Events.scala

```diff
@@ -0,0 +1,59 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.openwhisk.utils
+
+import akka.actor.ActorRef
+import akka.event.EventBus
+import akka.event.LookupClassification
+import java.time.Instant
+
+case class NodeStats(mem: Double, cpu: Double, ports: Int, lastSeen: Instant)
+abstract sealed trait OWEvent
+case class NodeStatsUpdate(stats: Map[String, NodeStats])
+final case class MsgEnvelope(topic: OWEvent, payload: Any)
+
+case object NodeStatsUpdateEvent extends OWEvent
+
+/**
+ * Publishes the payload of the MsgEnvelope when the topic of the
+ * MsgEnvelope equals the OWEvent specified when subscribing.
+ */
+object Events extends EventBus with LookupClassification {
```

Comment:
We can possibly use the [EventStream][1] of the `ActorSystem` itself. Should meet our requirement

[1]: https://doc.akka.io/docs/akka/2.5/event-bus.html#event-stream

Reply:
Sounds good! will look into making this change